README.md

```diff
@@ -559,6 +560,24 @@ class Travis implements CiProvider {
 }
 ```
 
+### Assertions
+
+*keywords: invariant*
+
+Simple runtime assertion that narrows involved types using [assertion functions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions).
+
+Note: This function is not purely type level and leaves minimal runtime trace in generated code.
+
+```typescript
+const something: string | undefined = "abc" as any
+assert(something, "Something has to be defined!")
+// from now on `something` is string, if this wouldn't be a case, assert would throw
+
+const anything: any = "abc" as any
```

Comment:
isn't `...: any = ... as any` redundant? (I mean, specifying `any` once should be sufficient)